<tt>
&lt;div&nbsp;dir=&quot;ltr&quot;&gt;g'day&nbsp;folks,&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;just&nbsp;been&nbsp;rummaging&nbsp;through&nbsp;the&nbsp;rabbit&nbsp;mq&nbsp;tutorials&nbsp;and&nbsp;got&nbsp;myself&nbsp;stuck&nbsp;in&nbsp;a&nbsp;slight&nbsp;rut.&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;i'm&nbsp;perusing&nbsp;the&nbsp;pub/sub&nbsp;example&nbsp;in&nbsp;python&nbsp;on&nbsp;windows.&nbsp;&lt;/div&gt;&lt;div&gt;i've&nbsp;declared&nbsp;an&nbsp;exchange,&nbsp;and&nbsp;having&nbsp;run&nbsp;the&nbsp;script,&nbsp;I&nbsp;can&nbsp;see&nbsp;it's&nbsp;been&nbsp;created&nbsp;through&nbsp;the&nbsp;management&nbsp;console,&lt;/div&gt;&lt;div&gt;but&nbsp;I&nbsp;keep&nbsp;getting&nbsp;an&nbsp;exception&nbsp;as&nbsp;soon&nbsp;as&nbsp;I&nbsp;run&nbsp;either&nbsp;the&nbsp;producer&nbsp;or&nbsp;the&nbsp;consumer&nbsp;app.&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;if&nbsp;I&nbsp;comment&nbsp;out&nbsp;the&nbsp;exchange_declare&nbsp;line,&nbsp;all&nbsp;is&nbsp;well.&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;the&nbsp;logger&nbsp;says&nbsp;I&nbsp;cannot&nbsp;redeclare&nbsp;an&nbsp;exchange&nbsp;with&nbsp;a&nbsp;different&nbsp;type.&lt;/div&gt;&lt;div&gt;in&nbsp;the&nbsp;management&nbsp;window&nbsp;I&nbsp;can&nbsp;see&nbsp;it&nbsp;is&nbsp;of&nbsp;type&nbsp;fanout.&nbsp;the&nbsp;code&nbsp;is&nbsp;declaring&nbsp;it&nbsp;as&nbsp;fanout.&lt;/div&gt;&lt;div&gt;I&nbsp;was&nbsp;under&nbsp;the&nbsp;impression&nbsp;exchange_declare&nbsp;was&nbsp;idempotent.&nbsp;is&nbsp;this&nbsp;correct?&lt;/div&gt;&lt;/div&gt;
</tt>
